aea/decision_maker/scaffold.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains a scaffold of the decision maker class and auxilliary classes."""
+
+from typing import List
+
+from aea.decision_maker.base import DecisionMaker as BaseDecisionMaker
+from aea.decision_maker.base import LedgerStateProxy as BaseLedgerStateProxy
+from aea.decision_maker.base import OwnershipState as BaseOwnershipState
+from aea.decision_maker.base import Preferences as BasePreferences
+from aea.decision_maker.messages.base import InternalMessage
+from aea.decision_maker.messages.transaction import TransactionMessage
+
+
+class OwnershipState(BaseOwnershipState):
+    """Represent the ownership state of an agent."""
+
+    def __init__(self):
+        """
+        Instantiate an ownership state object.
+
+        :param decision_maker: the decision maker
+        """
+        raise NotImplementedError
+
+    def set(self, **kwargs,) -> None:
+        """
+        Set values on the ownership state.
+
+        :param amount_by_currency_id: the currency endowment of the agent in this state.
+        :param quantities_by_good_id: the good endowment of the agent in this state.
+        """
+        raise NotImplementedError
+
+    def apply_delta(self, **kwargs,) -> None:
+        """
+        Apply a state update to the ownership state.
+
+        This method is used to apply a raw state update without a transaction.
+
+        :param
+        :return: None
+        """
+        raise NotImplementedError
```

Comment:
Note, these two methods should only be used from the decision maker not the skills. Current interface does not make that clear at all.